app/src/main/java/org/schabi/newpipe/util/potoken/JavaScriptUtil.kt

```diff
@@ -0,0 +1,113 @@
+package org.schabi.newpipe.util.potoken
+
+import com.grack.nanojson.JsonObject
+import com.grack.nanojson.JsonParser
+import com.grack.nanojson.JsonWriter
+import okio.ByteString.Companion.decodeBase64
+import okio.ByteString.Companion.toByteString
+
+/**
+ * Parses the raw challenge data obtained from the Create endpoint and returns an object that can be
+ * embedded in a JavaScript snippet.
+ */
+fun parseChallengeData(rawChallengeData: String): String {
+    val scrambled = JsonParser.array().from(rawChallengeData)
+
+    val challengeData = if (scrambled.size > 1 && scrambled.isString(1)) {
+        val descrambled = descramble(scrambled.getString(1))
+        JsonParser.array().from(descrambled)
+    } else {
+        scrambled.getArray(1)
```

Comment:
I think this should be accessing index `0` and not `1`? At least it did so in the [previous JS version](https://github.com/Stypox/NewPipe/blob/2b183a057651a79aa826a110e3eebf126befdd6c/app/src/main/assets/po_token.html#L100).